src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionDiagnosticAnalyzer.cs

```diff
@@ -98,14 +109,14 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
                 return;
             }
 
-            if (IsWrittenAfter(semanticModel, local, block, anonymousFunction, cancellationToken))
+            var delegateType = semanticModel.GetTypeInfo(anonymousFunction, cancellationToken).ConvertedType as INamedTypeSymbol;
+            if (!delegateType.IsDelegateType() ||
+                delegateType.DelegateInvokeMethod == null)
             {
                 return;
             }
```

Comment:
this code was already here.  i just swapped it with the complex check which now comes afterwards.